incubator/hnc/pkg/controllers/setup.go

```diff
@@ -61,3 +79,53 @@ func createObjectReconciler(newObjectController bool, mgr ctrl.Manager, f *fores
 	}
 	return or, or.SetupWithManager(mgr)
 }
+
+// ReconcileCounter counts total reconciles happened and current ongoing reconciles.
+type ReconcileCounter struct {
+	Log logr.Logger
+
+	// Total number of the object reconciles happened.
+	ObjectTotal int
+
+	// The number of current ongoing reconciles.
+	ObjectCur int
+
+	// Total number of the hier reconciles happened.
+	HierTotal int
+}
+
+// ObjectTotalCount counts the total number of object reconciles.
+func (counter *ReconcileCounter) ObjectTotalCount(log logr.Logger, objTotal <-chan int) {
+	for ot := range objTotal {
+		counter.ObjectTotal += ot
+		log.Info("Total number of object reconciles", "Total", counter.ObjectTotal)
+	}
+}
+
+// ObjectCurCount counts the current number of active object reconciles.
+func (counter *ReconcileCounter) ObjectCurCount(log logr.Logger, objCur <-chan int) {
+	for oc := range objCur {
+		counter.ObjectCur += oc
+		log.Info("Current number of object reconciles", "Current", counter.ObjectCur)
+	}
+}
+
+// HierTotalCount counts the total number of reconciles.
+func (counter *ReconcileCounter) HierTotalCount(log logr.Logger, hierTotal <-chan int) {
+	for t := range hierTotal {
+		counter.HierTotal += t
+		log.Info("Total number of hier reconciles", "Total", counter.HierTotal)
+	}
+}
+
+// This function is only used by reconcile counters for performance testing. We explicitly exclude the
+// default namespaces, so that only reconciles on objects in the user-created namespaces are counted.
+func isExcludedNS(ns string) bool {
+	ex := []string{"kube-system", "default", "cert-manager", "hnc-system", "kube-node-lease", "kube-public"}
```

Comment:
Can we make this a bit more minimal? Eg if we just exclude `kube-system` and `hnc-system`, will it still work?

Also, you could make this into a map:

```
var ex = map[string]bool{
  "kube-system": true,
  "hnc-system": true,
}

func isExcludedNS(ns string) bool {
  return ex[ns]
}
```

Although at that point you don't even really need `isExcludedNS` anymore :)

Reply:
Yes, it worked. Other namespaces don't have constantly changing objects.